packages/models-library/src/models_library/generics.py

```diff
@@ -1,66 +1,66 @@
 from collections.abc import ItemsView, Iterable, Iterator, KeysView, ValuesView
 from typing import Any, Generic, TypeVar
 
-from pydantic.generics import GenericModel
+from pydantic import BaseModel, RootModel
 
 DictKey = TypeVar("DictKey")
 DictValue = TypeVar("DictValue")
 
 
-class DictModel(GenericModel, Generic[DictKey, DictValue]):
-    __root__: dict[DictKey, DictValue]
+class DictModel(RootModel[dict[DictKey, DictValue]], Generic[DictKey, DictValue]):
```

Comment:
if ListModel and DictModel are not used anywhere in the  codebase, I suggest to drop them and rather use explicit `list[MyModel]` and `dict[str, MyModel]` CC @sanderegg , @GitHK @matusdrobuliak66